packages/Webkul/Shop/src/Http/Controllers/CartController.php

```diff
@@ -173,16 +173,26 @@ public function applyCoupon()
 
         try {
             if (strlen($couponCode)) {
-                Cart::setCouponCode($couponCode)->collectTotals();
-
-                if (Cart::getCart()->coupon_code == $couponCode) {
-                    return response()->json([
-                        'success' => true,
-                        'message' => trans('shop::app.checkout.total.success-coupon'),
-                    ]);
+                $validateStatus = $this->validateCouponStatus($couponCode);
```

Comment:
This can be done like below

```
$coupon = $this->cartRuleCouponRepository->findOneByField('code', $couponCode);

if ($coupon->cart_rule->status) {
     ....
}
```

Reply:
Updated.